woocommerce/facebook-for-woocommerce
# Add filter to allow changing the Facebook Retailer ID.

### Changes proposed in this Pull Request:

This adds a filter that allows changing of the Facebook Retailer ID for a product. Some other Facebook plugins that also operate on Retailer IDs generate the ids using different logic. If the ids don't match Facebook can't match certain events and products. This filter allows adding of a code that would generate the same ids for the products as 3rd party plugins.

This replaces https://github.com/woocommerce/facebook-for-woocommerce/pull/1197 the code is the same with an exception od using different filter name ( to match other filters in the product ) and updated documentation.

_Replace this with a good description of your changes & reasoning._


### Screenshots:




### Detailed test instructions:


1. Sync products.
2. Check products in the logs and observe the ids.
3. Use the filter to modify the ids, for example ( use WC product id for FB id ):

```php
add_filter( 'wc_facebook_fb_retailer_id', 'fb_id_filter', 10, 2 );

function fb_id_filter( $fb_retailer_id, $woo_product ) {
	return (string) $woo_product->get_id();
}
```
4. Sync, check the logs, and compare the products ids.


### Changelog entry

> Add - Filter to change Facebook Retailer ID, wc_facebook_fb_retailer_id.
